### PR TITLE
Allow to disable user selection in the grid model.

### PR DESCRIPTION
**Pull request by denis-anisimov**
Fixes vaadin/framework8-issues#546

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/framework/8345)

### CODE REVIEW

**Comment by caalador** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java, line 38 at r1](https://reviewable.io:443/reviews/vaadin/framework/8345#-KbUKMWJJ8eqgXAU3538:-KbUKMWJJ8eqgXAU3539:b-h2zmvm) ([raw file](https://github.com/vaadin/framework/blob/ad892468f79fa94928a5edfb7112ca48bff00894/uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java#L38)):*
> ```Java
>         checkSelection(1);
> 
>         $(ButtonElement.class).first().click();
> ```

Could have a comment that we change selectionModel from single select to multi select.
Test is hard to follow without looking at how the TestUI is built.

---

*[uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java, line 44 at r1](https://reviewable.io:443/reviews/vaadin/framework/8345#-KbUJyGxT5gYv42BwOlT:-KbUJyGxT5gYv42BwOlU:b73huah) ([raw file](https://github.com/vaadin/framework/blob/ad892468f79fa94928a5edfb7112ca48bff00894/uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java#L44)):*
> ```Java
>     }
> 
>     private void checkSelection(int index) {
> ```

Would rename to be more descriptive of actual check as now it seems like the row should be selected.
For instance assertNoSelection or  checkRowNotSelected.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8345)*

**Comment by denis-anisimov** [comment]
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java, line 38 at r1](https://reviewable.io:443/reviews/vaadin/framework/8345#-KbUKMWJJ8eqgXAU3538:-KbUWE3Fz66bm1ycE5T5:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad892468f79fa94928a5edfb7112ca48bff00894/uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java#L38)):*
<details><summary><i>Previously, caalador wrote…</i></summary><blockquote>

Could have a comment that we change selectionModel from single select to multi select.
Test is hard to follow without looking at how the TestUI is built.
</blockquote></details>

Done.

---

*[uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java, line 44 at r1](https://reviewable.io:443/reviews/vaadin/framework/8345#-KbUJyGxT5gYv42BwOlT:-KbUWTwAGuGkT7eQAvWl:b-896fix) ([raw file](https://github.com/vaadin/framework/blob/ad892468f79fa94928a5edfb7112ca48bff00894/uitest/src/test/java/com/vaadin/tests/components/grid/GridDisallowUserSelectionTest.java#L44)):*
<details><summary><i>Previously, caalador wrote…</i></summary><blockquote>

Would rename to be more descriptive of actual check as now it seems like the row should be selected.
For instance assertNoSelection or  checkRowNotSelected.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8345)*

**Comment by caalador** [comment]
:lgtm:

---

Review status: 6 of 7 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8345#-:-KbU_fktUJPPwDYOnPOU:bnfp4nl)*

**Comment by denis-anisimov** [comment]
The test file is not marked as reviewed.
Could you please mark it ?

---

Review status: 6 of 7 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8345#-:-KbUa_0XlVc8oeACr0Ku:b-oo9jpv)*

**Comment by denis-anisimov** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/framework/8345)*
